pilot/cmd/pilot-agent/status/util/stats.go

```diff
@@ -24,45 +24,66 @@ import (
 )
 
 const (
-	statCdsUpdatesSuccess   = "cluster_manager.cds.update_success"
-	statCdsUpdatesRejection = "cluster_manager.cds.update_rejected"
-	statLdsUpdatesSuccess   = "listener_manager.lds.update_success"
-	statLdsUpdatesRejection = "listener_manager.lds.update_rejected"
+	statCdsVersion    = "cluster_manager.cds.version"
+	statLdsVersion    = "listener_manager.lds.version"
+	statServerState   = "server.state"
+	versionStatsRegex = "^(cluster_manager.cds|listener_manager.lds).version$"
 )
 
+type stat struct {
+	name  string
+	value *uint64
+	found bool
+}
+
 // Stats contains values of interest from a poll of Envoy stats.
 type Stats struct {
-	CDSUpdatesSuccess   uint64
-	CDSUpdatesRejection uint64
-	LDSUpdatesSuccess   uint64
-	LDSUpdatesRejection uint64
+	// Hash of the contents from the last successful cluster update.
+	CDSVersion uint64
+	// Hash of the contents from the last successful listener update.
+	LDSVersion uint64
+	// Server State of Envoy.
+	ServerState uint64
 }
 
 // String representation of the Stats.
 func (s *Stats) String() string {
-	return fmt.Sprintf("cds updates: %d successful, %d rejected; lds updates: %d successful, %d rejected",
-		s.CDSUpdatesSuccess,
-		s.CDSUpdatesRejection,
-		s.LDSUpdatesSuccess,
-		s.LDSUpdatesRejection)
+	return fmt.Sprintf("cds version: %d , lds version: %d",
+		s.CDSVersion,
+		s.LDSVersion)
 }
 
-// GetStats from Envoy.
-func GetStats(localHostAddr string, adminPort uint16) (*Stats, error) {
-	input, err := doHTTPGet(fmt.Sprintf("http://%s:%d/stats?usedonly", localHostAddr, adminPort))
+// GetServerState returns the current Envoy state by checking the "server.state" stat.
+func GetServerState(localHostAddr string, adminPort uint16) (*uint64, error) {
+	stats, err := doHTTPGet(fmt.Sprintf("http://%s:%d/stats?usedonly&filter=%s", localHostAddr, adminPort, statServerState))
+	if err != nil {
+		return nil, multierror.Prefix(err, "failed retrieving Envoy server state stat:")
+	}
+
+	s := &Stats{}
+	allStats := []*stat{
```

Comment:
What are the stats value when server is not alive?

Reply:
Here are the other values https://github.com/envoyproxy/envoy/blob/524549f2c5a45a0cc8130c4f7d5130254d898506/api/envoy/admin/v2alpha/server_info.proto#L16